migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/MisoServiceManager.java

```diff
@@ -301,1407 +439,935 @@ public void setUpSecurityContext(User migrationUser) {
     SecurityContextHolder.setContext(context);
   }
 
-  // TODO: Add naming scheme config instead of hard-coding
   public NamingScheme getNamingScheme() {
-    if (namingScheme == null) {
-      namingScheme = new OicrNamingScheme();
+    return getService(NamingScheme.class);
+  }
+
+  private <T> T getService(Class<T> clazz) {
+    @SuppressWarnings("unchecked")
+    T service = (T) services.get(clazz);
```

Comment:
You can remove the unchecked warning by doing:
```
T service = clazz.cast(services.get(clazz));
```

Reply:
Interesting... I feel like `Class#cast` should cause the same compiler warning. Changed anyway :man_shrugging: